httpbakery/form/form.go

```diff
@@ -117,48 +82,48 @@ type LoginBody struct {
 	Form map[string]interface{} `json:"form"`
 }
 
+// Visitor implements httpbakery.Visitor
+// by providing form-based interaction.
+type Visitor struct {
+	// Filler holds the form filler that will be used when
+	// form-based interaction is required.
+	Filler form.Filler
+}
+
 // visitWebPage performs the actual visit request. It attempts to
 // determine that form login is supported and then download the form
 // schema. It calls v.handler.Handle using the downloaded schema and then
 // submits the returned form. Any error produced by v.handler.Handle will
 // not have it's cause masked.
-func (v webPageVisitor) visitWebPage(u *url.URL) error {
-	req, err := http.NewRequest("GET", u.String(), nil)
-	if err != nil {
-		return errgo.Notef(err, "cannot create request")
-	}
-	req.Header.Set("Accept", "application/json")
-	var lm loginMethods
-	if err := v.client.Do(req, nil, &lm); err != nil {
-		if v.fallback != nil {
-			if err := v.fallback(u); err != nil {
-				return errgo.Mask(err, errgo.Any)
-			}
-			return nil
-		}
-		return errgo.Notef(err, "cannot get login methods")
+func (v Visitor) VisitWebPage(client *httpbakery.Client, methodURLs map[string]*url.URL) error {
+	return v.visitWebPage(client, methodURLs)
```

Comment:
I think it's reasonable that we use the bakery client to do the interaction.
Can you think of a reason why we wouldn't want to?